networkit/cpp/randomization/CurveballImpl.cpp

```diff
@@ -154,29 +154,35 @@ void CurveballMaterialization::toGraphParallel(Graph &G) {
         }
     });
 
-    G.outEdges.swap(new_outEdges);
+    G.removeAllEdges();
+    for (index u = 0; u < new_outEdges.size(); u++) {
+        for (index j = 0; j < new_outEdges[u].size(); j++) {
+            G.addPartialEdge(unsafe, u, new_outEdges[u][j]);
+        }
+    }
 
     // Reserve the space
-    G.parallelForNodes([&](node v) { G.outEdges[v].reserve(outDeg[v] + missingEdgesCounts[v]); });
+    G.parallelForNodes(
+        [&](node v) { G.preallocateUndirected(v, outDeg[v] + missingEdgesCounts[v]); });
```

Comment:
The `addPartialEdge` loop can be moved into the parallelForNodes below (after preallocation).